src/components/FileSelector/index.js

```diff
@@ -0,0 +1,320 @@
+import React, { useRef, useState, useEffect, useImperativeHandle } from 'react';
+import PropTypes from 'prop-types';
+import Label from '../Input/label';
+import useReduxForm from '../../libs/hooks/useReduxForm';
+import RenderIf from '../RenderIf';
+import containsFiles from './helpers/containsFiles';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import {
+    StyledContainer,
+    StyledDropzone,
+    StyledInput,
+    TruncatedText,
+    StyledBackdrop,
+    StyledIconContainer,
+    StyledButtonIcon,
+} from './styled';
+import { UploadIcon, ErrorIcon, FileIcon, FilesIcon, CancelIcon } from './icons';
+import { useUniqueIdentifier, useErrorMessageId, useLabelId } from '../../libs/hooks';
+
+const FileSelector = React.forwardRef((props, ref) => {
+    const {
+        className,
+        style,
+        id,
+        name,
+        label,
+        error,
+        bottomHelpText,
+        placeholder,
+        tabIndex,
+        required,
+        multiple,
+        disabled,
+        variant,
+        hideLabel,
+        accept,
+        onChange,
+        onFocus,
+        onBlur,
+    } = useReduxForm(props);
+
+    const [isDragOver, setIsDragOver] = useState(false);
+    const [files, setFiles] = useState();
+    const [hasFocus, setHasFocus] = useState();
+
+    const inputId = useUniqueIdentifier('input');
+    const buttonId = useUniqueIdentifier('cancel-button');
+    const dropzoneId = useUniqueIdentifier('dropzone');
+    const labelId = useLabelId(label);
+    const errorMessageId = useErrorMessageId(error);
+
+    const inputRef = useRef();
+    useImperativeHandle(ref, () => ({
+        focus: () => {
+            inputRef.current.focus();
+        },
+        click: () => {
+            inputRef.current.click();
+        },
+        blur: () => {
+            inputRef.current.blur();
+        },
+    }));
```

Comment:
I think you don't need to define "inputRef", that you can the "ref" that comes by parameters

Reply:
I need to define inputRef because even when the developer does not pass a ref I need to send click events to the input and set the files

Reply:
@HellWolf93 You can try using the "ref" that comes by parameters, I am working on a component that is the same situation and it works for me with ref. It seems that forwardRef always sends a reference